virtual_ecosystem/models/abiotic/abiotic_model.py

```diff
@@ -0,0 +1,347 @@
+            ),
+        )
+
+        initial_conductivities = conductivities.initialise_conductivities(
+            layer_heights=self.data["layer_heights"],
+            initial_air_conductivity=self.model_constants.initial_air_conductivity,
+            top_leaf_vapour_conductivity=(
+                self.model_constants.top_leaf_vapour_conductivity
+            ),
+            bottom_leaf_vapour_conductivity=(
+                self.model_constants.bottom_leaf_vapour_conductivity
+            ),
+            top_leaf_air_conductivity=self.model_constants.top_leaf_air_conductivity,
+            bottom_leaf_air_conductivity=(
+                self.model_constants.bottom_leaf_air_conductivity
+            ),
+        )
+
+        # Update data object
+        for output_dict in (
+            initial_microclimate,
+            initial_canopy_and_soil,
+            initial_conductivities,
+        ):
+            self.data.add_from_dict(output_dict=output_dict)
+
+    def spinup(self) -> None:
+        """Placeholder function to spin up the abiotic model."""
+
+    def update(self, time_index: int, **kwargs: Any) -> None:
+        """Function to update the abiotic model.
+
+        The function updates the microclimate in the following order:
+
+        * wind profiles
+        * soil energy balance
+        * conductivities
+        * canopy energy balance for each layer
+        * TODO add all soil fluxes to atmosphere
+        * TODO update soil temperatures
+
+        Args:
+            time_index: The index of the current time step in the data object.
+        """
+
+        # TODO This selection of layers should be included in LayerStructure at the
+        # start of the simulation and updated at each time step (except topsoil index)
+        # At the moment this is duplicated in setup() and other parts of the Virtual
+        # Ecosystem
+        true_canopy_indexes = (
+            self.data["leaf_area_index"][
+                self.data["leaf_area_index"]["layer_roles"] == "canopy"
+            ]
+            .dropna(dim="layers", how="all")
+            .indexes["layers"]
+        )
+        true_canopy_layers_n = len(true_canopy_indexes)
+        empty_canopy_layers = self.layer_structure.canopy_layers - true_canopy_layers_n
+        topsoil_layer_index = self.layer_structure.layer_roles.index("soil")
+
+        # Wind profiles
+        wind_update_inputs: dict[str, DataArray] = {}
+
+        for var in ["layer_heights", "leaf_area_index", "air_temperature"]:
+            selection = (
+                self.data[var]
+                .where(self.data[var].layer_roles != "soil")
+                .dropna(dim="layers")
+            )
+            wind_update_inputs[var] = selection
+
+        wind_update = wind.calculate_wind_profile(
+            canopy_height=self.data["canopy_height"].to_numpy(),
+            wind_height_above=(self.data["canopy_height"] + 15).to_numpy(),
```

Comment:
This `15` seems a bit arbitrary. An explanation of why this number will be useful.

Reply:
I agree, that is why there is still a TODO hanging around. I will address this in a separate PR, the wind component needs a few other fixes, too...

Reply:
Sorry, didn't see the TODO. Good, then!